### PR TITLE
Be able to map multi-relay to single native address

### PR DESCRIPTION
**Pull request by girazoki**
Support multi Relay to single Native by adding different rewards

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -449,7 +449,20 @@ pub mod pallet {
 				total_contributors += 1;
 
 				if let Some(native_account) = native_account {
-					AccountsPayable::<T>::insert(native_account, reward_info);
+					if let Some(inserted_reward_info) = AccountsPayable::<T>::get(native_account) {
+						// the native account has already some rewards in, we add the new ones
+						AccountsPayable::<T>::insert(
+							native_account,
+							RewardInfo {
+								total_reward: inserted_reward_info.total_reward
```
Comment (crystalin) [inline src/lib.rs:457]: I think we allow changing the reward account @girazoki.
What would happen in that case?

**Review by crystalin — APPROVED** [review]

File: src/lib.rs (hunk 1 above)
Comment (girazoki) [reply src/lib.rs:457]: We allow changing the moonbeam address only. Two scenarios:

- The change of the reward account happens when the reward_vec has not been fully initialized. In that case, the reward accumulated up to that point gets moved to the new address, while you can still continue to initialize the previous address (obvioulsy, starting from 0 again). So say you have address A with X rewards initialized, you call `update_reward_address` to change it to B. In this case A has 0 rewards while B has X. If you later re-initialize A with Y rewards, A will have Y and B will have X.  

- The change of the reward account happens after we have fully initialize the block. In that case, all the accumulated reward moves to the new address.

Comment (crystalin) [reply src/lib.rs:457]: So when 2 rewards go to the same account, changing the account will apply to both rewards right ?

Comment (girazoki) [reply src/lib.rs:457]: Correct, I will add a test for that to make us feel safier

**Pull request merged**